libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2338,6 +2344,47 @@ bool TypeChecker::visit(IndexAccess const& _access)
 	return false;
 }
 
+bool TypeChecker::visit(IndexRangeAccess const& _access)
+{
+	_access.baseExpression().accept(*this);
+	TypePointer exprType = type(_access.baseExpression());
+	ArrayType const* arrayType = nullptr;
+
+	Expression const& start = _access.startExpression();
+	Expression const* end = _access.endExpression();
+
+	bool isLValue = false;
+	bool isPure = _access.baseExpression().annotation().isPure && start.annotation().isPure;
+
+	expectType(start, *TypeProvider::uint256());
+	if (end)
+	{
+		expectType(*end, *TypeProvider::uint256());
+		if (!end->annotation().isPure)
+			isPure = false;
+	}
+
+	if (exprType->category() == Type::Category::TypeType)
+	{
+		m_errorReporter.typeError(_access.location(), "Types cannot be sliced.");
+		_access.annotation().type = exprType;
+		return false;
+	}
+	else if (auto const* arraySlice = dynamic_cast<ArraySliceType const*>(exprType))
+		arrayType = &arraySlice->arrayType();
+	else if (!(arrayType = dynamic_cast<ArrayType const*>(exprType)))
+		m_errorReporter.fatalTypeError(_access.location(), "Index range access is only possible for arrays and array slices.");
```

Comment:
The equivalent in ``IndexAccess`` is a fatal error as well... we could make both of them non-fatal and just assign ``_access.annotation().type = exprType`` here - that might lead to confusing results later, but probably not even too bad ones...